src/lib/coda_base/sparse_ledger.ml

```diff
@@ -134,9 +134,15 @@ let get_or_initialize_exn account_id t idx =
   let account = get_exn t idx in
   if Public_key.Compressed.(equal empty account.public_key) then
     let public_key = Account_id.public_key account_id in
+    let token_id = Account_id.token_id account_id in
+    let delegate =
+      (* Only allow delegation if this account is for the default token. *)
+      if Token_id.(equal default) token_id then public_key
+      else Public_key.Compressed.empty
+    in
     ( `Added
     , { account with
-        delegate= public_key
+        delegate
```

Comment:
It seems a bit funny to use the `empty` key to indicate no delegation.

Would it be better to use an option?

Reply:
Actually, haven't we been using the account pk for the delegate in the case where there's no delegation?

Reply:
Yeah that's the default (all the accounts delegate to themselves?). But in the case of non-default tokens we don't want to allow delegation at all